libflux/src/flux/semantic/flatbuffers/types.rs

```diff
@@ -204,6 +225,39 @@ where
     mapped
 }
 
+fn build_env<'a>(
+    builder: &mut flatbuffers::FlatBufferBuilder<'a>,
+    env: Environment,
+) -> flatbuffers::WIPOffset<fb::TypeEnvironment<'a>> {
+    let assignments = build_vec(
+        env.values.into_iter().collect(),
+        builder,
+        build_type_assignment,
+    );
+    let assignments = builder.create_vector(assignments.as_slice());
+    fb::TypeEnvironment::create(
+        builder,
+        &fb::TypeEnvironmentArgs {
+            assignments: Some(assignments),
+        },
+    )
+}
+
+fn build_type_assignment<'a>(
+    builder: &mut flatbuffers::FlatBufferBuilder<'a>,
+    assignment: (String, PolyType),
+) -> flatbuffers::WIPOffset<fb::TypeAssignment<'a>> {
+    let id = builder.create_string(&assignment.0);
+    let ty = build_polytype(builder, assignment.1);
+    fb::TypeAssignment::create(
+        builder,
+        &fb::TypeAssignmentArgs {
+            id: Some(id),
+            ty: Some(ty),
+        },
+    )
+}
+
```

Comment:
It would be good to have a round-trip test for the type env.

Reply:
👍 I'll include one.